src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -2913,7 +2913,7 @@ trait Types
     //TODO this may be generalised so that the only constraint is dependencies are acyclic
     def approximate: MethodType = MethodType(params, resultApprox)
 
-    override def safeToString = paramString(this) + resultType
+    override def safeToString = s"${paramString(this)}: $resultType"
```

Comment:
I think you need to check if `resultType` is a `MethodType` and not add the colon if so. Maybe look at the "object creation impossible" error in `RefChecks` for how it's done there.